rocks-lib/src/lua_package/mod.rs

```diff
@@ -14,11 +14,14 @@ pub struct LuaPackage {
 }
 
 impl LuaPackage {
-    pub fn new(name: String, version: String) -> Result<Self> {
-        Ok(Self {
-            name: PackageName::new(name),
-            version: PackageVersion::parse(&version)?,
-        })
+    pub fn new(name: PackageName, version: PackageVersion) -> Self {
+        Self { name, version }
+    }
+    pub fn parse(name: String, version: String) -> Result<Self> {
+        Ok(Self::new(
+            PackageName::new(name),
+            PackageVersion::parse(&version)?,
+        ))
     }
```

Comment:
question: any benefit from doing `impl From<(String, String)>` as a bonus, or is that too much "magic"?

Reply:
The implementation would have to panic if the version can't be parsed, as it doesn't return a `Result`.

Reply:
We have a `FromStr`, so I'd say yagni to `From` for now.